src/main/java/tigase/server/xmppclient/ClientConnectionManager.java

```diff
@@ -842,6 +848,9 @@ protected void processCommand(Packet packet) {
 					} catch (Exception e) {
 					}
 					serv.stop();
+					for (XMPPIOProcessor processor : processors) {
+						processor.serviceStopped(serv, true);

```

Comment:
is that guaranteed after stream closure commands, `StreamManagementIOProcessor` to send errors for queued packets? In the case where I reported the bug, somehow `XMPPIOService.forceStop` called `StreamManagementIOProcessor` to send errors for queued packets in `outQueue` before stream closure commands. The fix in pr looks like solves alignment between WaitingPackets and stream closure command but not the case originally reported.

Reply:
Actually, it solves both issues as we have created additional tests in TTS-NG to verify if messages are redelivered properly to the clients and those tests indicated issues in places which you reported. Then changes from PR were applied and that fixed issues with lost messages.

Please note, that it is expected for `message` processor to block redelivery of errors in many cases, ie. when another user connection for the same bare jid was active. This was common case in the detected errors as rebound  session (new session using same resource) was replacing old session but message redelivery was not instant but delayed due to missing call during closing of old session. This delay caused message to be dropped instead of being redelivered.